app/store.go

```diff
@@ -348,30 +347,19 @@ func (s *StoreApp) BeginBlock(req abci.RequestBeginBlock) (res abci.ResponseBegi
 // TODO: investigate response tags as of 0.11 abci
 func (s *StoreApp) EndBlock(_ abci.RequestEndBlock) (res abci.ResponseEndBlock) {
 	res.ValidatorUpdates = weave.ValidatorUpdatesToABCI(s.pending)
-	s.pending = nil
+	s.pending = weave.ValidatorUpdates{}
 	return
 }
 
 // AddValChange is meant to be called by apps on DeliverTx
 // results, this is added to the cache for the endblock changeset
 func (s *StoreApp) AddValChange(diffs []weave.ValidatorUpdate) {
-	// ensures multiple updates for one validator are combined into one slot
-	for _, d := range diffs {
-		idx := pubKeyIndex(d, s.pending)
-		if idx >= 0 {
-			s.pending[idx] = d
-		} else {
-			s.pending = append(s.pending, d)
-		}
-	}
-}
+	updateSlice := s.pending.ValidatorUpdates
```

Comment:
Why these two renaming lines? There is no concurrency to worry about in DeliverTx

```go
s.pending.ValidatorUpdates = append(s.pending.ValidatorUpdates, diffs...)
// and with api change proposed below....
s.pending = s.pending.Deduplicate(false)
```